meson.build

```diff
@@ -0,0 +1,125 @@
+project(
+  'spoa',
+  ['cpp'],
+  version : '4.1.0',
+  default_options : [
+    'buildtype=release',
+    'warning_level=2',
+    'cpp_std=c++11',
+    'b_ndebug=if-release'],
+  license : 'MIT',
+  meson_version : '>=0.60.0'
+)
+
+############
+# CXXFLAGS #
+############
+
+spoa_warning_flags = []
+cpp = meson.get_compiler('cpp')
+foreach cflag : [
+  '-Wall',
+  '-Wextra',
```

Comment:
`warning_level=3` should already do all of this, try it and then get rid of all of this